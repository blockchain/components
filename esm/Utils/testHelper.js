import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}