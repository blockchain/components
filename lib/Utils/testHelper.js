"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}