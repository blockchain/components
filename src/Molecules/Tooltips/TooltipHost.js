// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
// $FlowFixMe
import ReactTooltip from 'react-tooltip'
import styled from 'styled-components'

const Wrapper = styled.div`
  width: fit-content;
  display: inline-flex;
`

type PropsType = {
  +children: React.Node,
  +id: string,
  +tip?: string,
}

class TooltipHost extends React.PureComponent<PropsType> {
  componentDidMount() {
    ReactTooltip.rebuild()
  }

  render() {
    const { tip, children, id } = this.props
    const dataTip = tip == null ? '' : tip

    return (
      <Wrapper data-for={id} data-tip={dataTip} {...this.props}>
        {children}
      </Wrapper>
    )
  }
}

TooltipHost.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  tip: PropTypes.string,
}

export default TooltipHost
