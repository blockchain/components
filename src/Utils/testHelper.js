import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
