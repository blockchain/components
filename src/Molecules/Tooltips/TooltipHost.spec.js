import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import TooltipHost from './TooltipHost'

describe('TooltipHost', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(
      <TooltipHost id="storybook.test">TooltipHost</TooltipHost>,
    )

    expect(tree.find(TooltipHost)).toMatchSnapshot()
  })

  it('should render correctly with tip', () => {
    const tree = mountWithTheme(
      <TooltipHost id="storybook.test" tip="test">
        TooltipHost
      </TooltipHost>,
    )

    expect(tree.find(TooltipHost)).toMatchSnapshot()
  })
})
