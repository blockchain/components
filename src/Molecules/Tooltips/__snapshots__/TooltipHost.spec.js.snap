// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipHost should render correctly 1`] = `
.c0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<TooltipHost
  id="storybook.test"
  tip="test"
>
  <styled.div
    data-for="storybook.test"
    data-tip="test"
    id="storybook.test"
    tip="test"
  >
    <div
      className="c0"
      data-for="storybook.test"
      data-tip="test"
      id="storybook.test"
    >
      TooltipHost
    </div>
  </styled.div>
</TooltipHost>
`;
